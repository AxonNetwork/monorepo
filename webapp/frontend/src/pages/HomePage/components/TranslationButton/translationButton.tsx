import React from 'react';
import { withNamespaces, WithNamespaces as Ii18nProps } from 'react-i18next';

const floatingActionButtonConfig: Partial<M.FloatingActionButtonOptions> = {
  direction: 'top',
};

export class TranslationButton extends React.Component<Ii18nProps> {
  public componentDidMount() {
    const elems = document.querySelectorAll('.fixed-action-btn');
    M.FloatingActionButton.init(elems, floatingActionButtonConfig);
  }

  public changeLanguage = (lng: string) => {
    return () => {
      const { i18n } = this.props;
      i18n.changeLanguage(lng);
    };
  }

  public render() {
    return (
      <div className='fixed-action-btn vertical'>
        <a className='btn-floating btn-large red'>
          <i className='large material-icons'>translate</i>
        </a>
        <ul>
          <li><a className='btn-floating red' onClick={this.changeLanguage('en')}>en</a></li>
          <li><a className='btn-floating green' onClick={this.changeLanguage('zh')}>zh</a></li>
          <li><a className='btn-floating yellow darken-1' onClick={this.changeLanguage('jp')}>jp</a></li>
        </ul>
      </div>
    );
  }
}

export default withNamespaces('TranslationButton')(TranslationButton);
