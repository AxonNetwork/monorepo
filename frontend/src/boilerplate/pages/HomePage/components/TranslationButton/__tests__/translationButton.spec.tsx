import 'materialize-css';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';

import { TranslationButton } from '../translationButton';

const tMock = (k: string) => k;
const i18nMock: any = {
  addResourceBundle: () => { return; },
};

describe('TranslationButton', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <TranslationButton t={tMock} i18n={i18nMock} />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
