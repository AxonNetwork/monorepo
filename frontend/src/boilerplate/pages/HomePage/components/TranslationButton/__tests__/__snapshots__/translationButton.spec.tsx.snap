// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslationButton should renders correctly 1`] = `
<div
  className="fixed-action-btn vertical"
>
  <a
    className="btn-floating btn-large red"
  >
    <i
      className="large material-icons"
    >
      translate
    </i>
  </a>
  <ul>
    <li>
      <a
        className="btn-floating red"
        onClick={[Function]}
      >
        en
      </a>
    </li>
    <li>
      <a
        className="btn-floating green"
        onClick={[Function]}
      >
        zh
      </a>
    </li>
    <li>
      <a
        className="btn-floating yellow darken-1"
        onClick={[Function]}
      >
        jp
      </a>
    </li>
  </ul>
</div>
`;
